ls1intum/Artemis
# `General`: Add "Dismiss all" button to alerts overlay

### Checklist
#### General


#### Client

### Motivation and Context



It is currently not possible to dismiss all alerts at once. Instead, a user has to click the little "X" in the top right of each alert, which can be time-consuming, especially if you quickly want to access content covered by alerts. 
This was a suggestion in my thesis that I wanted to cover after I finished, and it should solve #5630 well enough.

### Description


Added a button at the top of the alert stack that allows to dismiss all alerts at once. It is only visible if there is more than one alert; if there is only one alert, one can use the X button with the same effort and the new button would just take up space unnecessarily.

### Steps for Testing

Prerequisites:
- 1 User

1. Provoke the app to spawn multiple alerts at once. I recommend testing this locally, and just adding something like this somewhere into the code: 
```js
setTimeout(() => this.alertService.info('hi'), 1000);
setTimeout(() => this.alertService.success('Test'), 2000);
```
2. Verify that you can dismiss all alerts at once if you click the button.
3. Verify that the button only appears if the are multiple active alerts
4. Verify that the button disappears if you close all alerts but one.

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots




https://user-images.githubusercontent.com/23171488/197762849-aaa34fa8-d801-4526-a9f9-8040ebdf1b7e.mov
